sfdc/rest_client.bal

```diff
@@ -0,0 +1,527 @@
+    }
+
+    //Basic CRUD
+    # Accesses records based on the specified object ID, can be used with external objects.
+    #
+    # + path - Resource path
+    # + return - JSON result if successful else Error occured
+    @display {label: "Get Record"}
+    isolated remote function getRecord(@display {label: "Resource Path"} string path) returns @tainted@display 
+    {label: "Result"} json|Error {
+        http:Response|error response = self.salesforceClient->get(path);
+        return checkAndSetErrors(response);
+    }
+
+    # Create records based on relevant object type sent with json record.
+    #
+    # + sObjectName - SObject name value
+    # + recordPayload - JSON record to be inserted
+    # + return - Created entity ID if successful else Error occured
+    @display {label: "Create Record"}
+    isolated remote function createRecord(@display {label: "SObject Name"} string sObjectName, @display 
+                                          {label: "Record Payload"} json recordPayload) returns @tainted@display 
+    {label: "Created Entity ID"} string|Error {
+        http:Request req = new;
+        string path = prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName]);
+        req.setJsonPayload(recordPayload);
+
+        http:Response|error response = self.salesforceClient->post(path, req);
+        json result = check checkAndSetErrors(response);
+        json|error resultId = result.id;
+        if (resultId is json) {
+            return resultId.toString();
+        } else {
+            return error Error(resultId.message());
+        }
+
+    }
+
+    # Update records based on relevant object id.
+    #
+    # + sObjectName - SObject name value
+    # + id - SObject id
+    # + recordPayload - JSON record to be updated
+    # + return - true if successful else false or Error occured
+    @display {label: "Update Record"}
+    isolated remote function updateRecord(@display {label: "SObject Name"} string sObjectName, @display 
+                                          {label: "SObject ID"} string id, 
+                                          @display {label: "Record Payload"} json recordPayload) returns @tainted@display 
+    {label: "Result"} Error? {
+        http:Request req = new;
+        string path = prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName, id]);
+        req.setJsonPayload(recordPayload);
+        http:Response|error response = self.salesforceClient->patch(path, req);
+        json result = check checkAndSetErrors(response, false);
+    }
+
+    # Delete existing records based on relevant object id.
+    #
+    # + sObjectName - SObject name value
+    # + id - SObject id
+    # + return - true if successful else false or Error occured
+    @display {label: "Delete Record"}
+    isolated remote function deleteRecord(@display {label: "SObject Name"} string sObjectName, @display 
+                                          {label: "SObject ID"} string id) returns @tainted@display {label: "Result"} 
+    Error? {
```

Comment:
Can we change the custom error in these places too? 
Formatting error

Reply:
This will be a method signature change. Is it possible to do?

Reply:
@abeykoon please share your thoughts

Reply:
As it is a breaking change , we'll not do that.